<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>svntogit/packages.git - Git clone of the 'packages' repository
</title>
<meta name='generator' content='cgit '/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://git.archlinux.org/svntogit/packages.git/atom/trunk/pygame-v4l.patch?h=packages/python-pygame' type='application/atom+xml'/>
<link rel='vcs-git' href='https://git.archlinux.org/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='git://git.archlinux.org/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='ssh://git.archlinux.org/srv/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
</head>
<body>
	<div id="archnavbar"><!-- Arch Linux global navigation bar -->
		<div id="archnavbarlogo">
			<p><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more"></a></p>
		</div>
		<div id="archnavbarmenu">
			<ul id="archnavbarlist">
				<li id="anb-home"><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
				<li id="anb-packages"><a href="http://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
				<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
				<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
				<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and follow bugs">Bugs</a></li>
				<li id="anb-aur"><a href="https://aur.archlinux.org/" title="Arch Linux User Repository">AUR</a></li>
				<li id="anb-download"><a href="http://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
			</ul>
		</div>
	</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'><a href='/'>index</a> : <a title='svntogit/packages.git' href='/svntogit/packages.git/'>svntogit/packages.git</a></td></tr>
<tr><td class='sub'>Git clone of the 'packages' repository
</td><td class='sub right'>Evangelos Foutras</td></tr></table>
<table class='tabs'><tr><td>
<a href='/svntogit/packages.git/?h=packages/python-pygame'>summary</a><a href='/svntogit/packages.git/refs/?h=packages/python-pygame'>refs</a><a href='/svntogit/packages.git/log/trunk/pygame-v4l.patch?h=packages/python-pygame'>log</a><a class='active' href='/svntogit/packages.git/tree/trunk/pygame-v4l.patch?h=packages/python-pygame'>tree</a><a href='/svntogit/packages.git/commit/trunk/pygame-v4l.patch?h=packages/python-pygame'>commit</a><a href='/svntogit/packages.git/diff/trunk/pygame-v4l.patch?h=packages/python-pygame'>diff</a><a href='/svntogit/packages.git/stats/trunk/pygame-v4l.patch?h=packages/python-pygame'>stats</a></td><td class='form'><form class='right' method='get' action='/svntogit/packages.git/log/trunk/pygame-v4l.patch'>
<input type='hidden' name='h' value='packages/python-pygame'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/svntogit/packages.git/tree/?h=packages/python-pygame'>root</a>/<a href='/svntogit/packages.git/tree/trunk?h=packages/python-pygame'>trunk</a>/<a href='/svntogit/packages.git/tree/trunk/pygame-v4l.patch?h=packages/python-pygame'>pygame-v4l.patch</a></div><div class='content'>blob: 3b5a2b9d2bf943ca7f0fbefadac437246ef9201e (<a href='/svntogit/packages.git/plain/trunk/pygame-v4l.patch?h=packages/python-pygame'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
</pre></td>
<td class='lines'><pre><code>diff -crB pygame-1.9.1release/Setup.in pygame-1.9.1release-v4lpatch//Setup.in
*** pygame-1.9.1release/Setup.in	Thu Jul  2 06:41:56 2009
<span class="hl kwb">--- pygame-1.9.1release-v4lpatch//Setup.in	Thu Mar 24 17:31:22 2011</span>
***************
*** 34,40 ****
  _numericsndarray src/_numericsndarray.c $(SDL) $(MIXER) $(DEBUG)
  movie src/movie.c $(SDL) $(SMPEG) $(DEBUG)
  scrap src/scrap.c $(SDL) $(SCRAP) $(DEBUG)
! _camera src/_camera.c src/camera_v4l2.c src/camera_v4l.c $(SDL) $(DEBUG)
  pypm src/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
  
  GFX = src/SDL_gfx/SDL_gfxPrimitives.c 
<span class="hl kwb">--- 34,40 ----</span>
  _numericsndarray src/_numericsndarray.c $(SDL) $(MIXER) $(DEBUG)
  movie src/movie.c $(SDL) $(SMPEG) $(DEBUG)
  scrap src/scrap.c $(SDL) $(SCRAP) $(DEBUG)
! _camera src/_camera.c src/camera_v4l2.c $(SDL) $(DEBUG)
  pypm src/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
  
  GFX = src/SDL_gfx/SDL_gfxPrimitives.c 
diff -crB pygame-1.9.1release/src/_camera.c pygame-1.9.1release-v4lpatch//src/_camera.c
*** pygame-1.9.1release/src/_camera.c	Sun Mar 15 20:30:41 2009
<span class="hl kwb">--- pygame-1.9.1release-v4lpatch//src/_camera.c	Thu Mar 24 16:58:18 2011</span>
***************
*** 160,179 ****
  {
  #if defined(__unix__)
      if (v4l2_open_device(self) == 0) {
!         if (v4l_open_device(self) == 0) {
!             v4l2_close_device(self);
!             return NULL;
!         } else {
!             self-&gt;camera_type = CAM_V4L;
!             if (v4l_init_device(self) == 0) {
!                 v4l2_close_device(self);
!                 return NULL;
!             }
!             if (v4l_start_capturing(self) == 0) {
!                 v4l2_close_device(self);
!                 return NULL;
!             }
!         }
      } else {
          self-&gt;camera_type = CAM_V4L2;
          if (v4l2_init_device(self) == 0) {
<span class="hl kwb">--- 160,167 ----</span>
  {
  #if defined(__unix__)
      if (v4l2_open_device(self) == 0) {
! 	v4l2_close_device(self);
!         return NULL;
      } else {
          self-&gt;camera_type = CAM_V4L2;
          if (v4l2_init_device(self) == 0) {
diff -crB pygame-1.9.1release/src/camera.h pygame-1.9.1release-v4lpatch//src/camera.h
*** pygame-1.9.1release/src/camera.h	Fri Oct 10 04:37:10 2008
<span class="hl kwb">--- pygame-1.9.1release-v4lpatch//src/camera.h	Thu Mar 24 16:44:32 2011</span>
***************
*** 39,45 ****
  
      #include &lt;asm/types.h&gt;          /* for videodev2.h */
  
<span class="hl kwb">-     #include &lt;linux/videodev.h&gt;</span>
      #include &lt;linux/videodev2.h&gt;
  #endif
  
<span class="hl kwb">--- 39,44 ----</span>
***************
*** 51,57 ****
  #define RGB_OUT 1
  #define YUV_OUT 2
  #define HSV_OUT 4
<span class="hl kwb">- #define CAM_V4L 1</span>
  #define CAM_V4L2 2
  
  struct buffer 
<span class="hl kwb">--- 50,55 ----</span>
***************
*** 111,118 ****
  int v4l2_close_device (PyCameraObject* self);
  int v4l2_open_device (PyCameraObject* self);
  
<span class="hl kwb">- /* internal functions specific to v4l */</span>
<span class="hl kwb">- int v4l_open_device (PyCameraObject* self);</span>
<span class="hl kwb">- int v4l_init_device(PyCameraObject* self);</span>
<span class="hl kwb">- int v4l_start_capturing(PyCameraObject* self);</span>
  #endif
<span class="hl kwb">--- 109,112 ----</span>
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class="foot" style="padding-left:1em;padding-right:1em;">
<p>Copyright &copy; 2002-2016 <a href="mailto:jvinet@zeroflux.org"
title="contact Judd Vinet">Judd Vinet</a> and <a href="mailto:aaron@archlinux.org"
title="contact Aaron Griffin">Aaron Griffin</a>. The Arch Linux name and logo
are recognized trademarks. Some rights reserved. The registered trademark
Linux&reg; is used pursuant to a sublicense from LMI, the exclusive licensee
of Linus Torvalds, owner of the mark on a world-wide basis.</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
